src/windy/platforms/win32/platform.nim

```diff
@@ -1047,6 +1047,40 @@ proc buttonReleased*(window: Window): ButtonView =
 proc buttonToggle*(window: Window): ButtonView =
   window.state.buttonToggle.ButtonView
 
+proc getClipboardImage*(): Image = 
+  init()
+  if IsClipboardFormatAvailable(CF_DIB) == FALSE:
+    return nil
+
+  if OpenClipboard(helperWindow) == 0:
+    return nil
+
+  let dataHandle = GetClipboardData(CF_DIB)
+  var bitmapData = newString(0)
+  if dataHandle != 0:
+    let p = cast[ptr BYTE](GlobalLock(dataHandle))
+    if p != nil:
+      bitmapData.add("BM") # The header field used to identify the BMP
```

Comment:
It is strange to me that you are creating a bitmap here. If you can just get pixels you could just create a pixie image. If you get a "file" don't you need to support all file types? 

Is strange that you are manually creating bitmap to just throw it a way by parsing it.

Reply:
> It is strange to me that you are creating a bitmap here. If you can just get pixels you could just create a pixie image. If you get a "file" don't you need to support all file types?


I create a bitmap from the memory windows gives the program because that's the format most likely to be in the clipboard it seems like (CF_DIB, see [here](https://docs.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats)). CF_DIB more specifically is a "a [BITMAPINFO](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfo) structure followed by the bitmap bits", which is essentially a BMP file without the file header ("BM", a couple zeros, and the offset to the pixels).

Since fully parsing the BITMAPINFO struct to create a pixie compatible image buffer seems to be complicated, and that work is already done in the pixie code to load a bmp file, I thought it would make sense to reuse that code.

This also seems to be how other platform layers accomplish this, see [here](https://github.com/python-pillow/Pillow/blob/95cff6e959bb3c37848158ed2145d49d49806a31/src/display.c#L501) for Pillow requesting the clipboard format in DIB, and then [here](https://github.com/python-pillow/Pillow/blob/95cff6e959bb3c37848158ed2145d49d49806a31/src/PIL/ImageGrab.py#L114) where it takes that image buffer and loads it as a "DibImageFile", and then finally [here](https://github.com/python-pillow/Pillow/blob/95cff6e959bb3c37848158ed2145d49d49806a31/src/PIL/BmpImagePlugin.py#L275) where the dib image file is treated as a bmp

Reply:
That does make sense, thank you for the explanation.